src/nlp/api.jl

```diff
@@ -215,6 +387,114 @@ jprod!(
   Jv::AbstractVector,
 ) = jprod!(nlp, x, v, Jv)
 
+"""
+    Jv = lin_jprod(nlp, x, v)
+
+Evaluate ``J(x)v``, the linear Jacobian-vector product at `x`.
+"""
+function lin_jprod(nlp::AbstractNLPModel, x::AbstractVector, v::AbstractVector)
+  @lencheck nlp.meta.nvar x v
+  Jv = similar(v, nlp.meta.nlin)
+  return lin_jprod!(nlp, x, v, Jv)
+end
+
+"""
+    Jv = lin_jprod!(nlp, x, v, Jv)
+
+Evaluate ``J(x)v``, the linear Jacobian-vector product at `x` in place.
+"""
+function lin_jprod! end
+
+"""
+    Jv = lin_jprod!(nlp, rows, cols, vals, v, Jv)
```

Comment:
What is the use of this variant?

Reply:
It is for cases where one needs to compute (or one has computed) the jacobian first to compute the product.

The same exists with `jprod!(nlp, rows, cols, vals, v, Jv)` and `nln_jprod!(nlp, rows, cols, vals, v, Jv)`.